src/librustc/ty/layout.rs

```diff
@@ -962,13 +963,11 @@ impl<'a, 'tcx> LayoutDetails {
             let mut inverse_memory_index: Vec<u32> = (0..fields.len() as u32).collect();
 
             // Anything with repr(C) or repr(packed) doesn't optimize.
+            let can_optimize = (repr.flags & ReprFlags::IS_UNOPTIMISABLE).is_empty();
```

Comment:
I think I prefer naming this `optimize`, making it mutable, and `&=`-ing it below instead.